cmds/dhclient/dhcp6client/dhcp6client.go

```diff
@@ -0,0 +1,128 @@
+package dhcp6client
+
+import (
+	"fmt"
+	"net"
+	"time"
+
+	"github.com/mdlayher/dhcp6"
+	// "golang.org/x/net/icmp"
+	//"golang.org/x/net/ipv6"
+	// "github.com/d2g/dhcp4"
+)
+
+type Client struct {
+	hardwareAddr  net.HardwareAddr //The HardwareAddr to send in the request.
+	ignoreServers []net.IP         //List of Servers to Ignore requests from.
+	timeout       time.Duration    //Time before we timeout.
+	broadcast     bool             //Set the Bcast flag in BOOTP Flags
+	connection    connection       //The Connection Method to use
+}
+
+/*
+*  * Abstracts the type of underlying socket used
+*   */
+type connection interface {
+	Close() error
+	Write(p *dhcp6.Packet, mac net.HardwareAddr) error
+	// WriteNeighborAd(src, dst net.IP, pb []byte) error
+	ReadFrom() ([]byte, error)
+	// SetReadTimeout(t time.Duration) error
+}
+
+func New(haddr net.HardwareAddr, conn connection, timeout time.Duration) (*Client, error) {
+	c := Client{
+		broadcast: true,
+	}
+
+	c.hardwareAddr = haddr
+	c.connection = conn
+	c.timeout = timeout
```

Comment:
you might want to try setting broadcast here and see if it removes the need for promiscuous interface setting.